pipeline.cpp

```diff
@@ -85,6 +85,32 @@ float crossProduct(Vertex* const vertex1, Vertex* const vertex2)
     return ((*vertex1).x * (*vertex2).y) - ((*vertex2).x * (*vertex1).y);
 }
 
+/*************************************************************
+ * INTERPOLATE
+ * Assign the appropriate weight to the pixel at it's x and
+ * y coordinate. The weight is determined by its distance
+ * from all three vertices
+ ************************************************************/
+double interoplate(double y, double x, Vertex* const triangle, double attr1, double attr2, double attr3)
+{
+    // % of weight that attribute 1 will be diluted
+    double w1 = ((triangle[1].y - triangle[2].y) * (x - triangle[2].x) + (triangle[2].x - triangle[1].x) * (y - triangle[2].y)) /
```

Comment:
Try cleaning this up if you can. This can use other variables in order to simplify it a bit more.